agent/exploits/cve_2024_45409.py

```diff
@@ -0,0 +1,46 @@
+"""Agent Asteroid implementation for CVE-2024-45409"""
+
+import re
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = "GitLab SAML Authentication Bypass Vulnerability."
+VULNERABILITY_REFERENCE = "CVE-2024-45409"
+VULNERABILITY_DESCRIPTION = (
+    "An issue was discovered in GitLab CE/EE affecting all versions "
+    "prior to 17.0.8, 17.1.8, 17.2.7, 17.3.3, and 16.11.10. "
+    "An unauthenticated attacker with access to any signed saml document (by the IdP) can thus forge a SAML Response/Assertion "
+    "with arbitrary contents. This would allow the attacker to log in as arbitrary user within the vulnerable system."
+)
+RISK_RATING = "POTENTIALLY"
+
+VERSIONS_HASHES = [
+    "b52d3dd3b307bb936ad862671cae67163327a698f3dc2c4f232173d621af2a87",
+    "487de60d9ab3205ffe87ebd3b4dad8a80e7fa602d5390d64849d67476445b683",
+    "e2dd787585a2324e8c33f0c7f58327936bd32309cda3f49932b8fd439522e77b",
+    "22918e5e48d718e0977422d6c63347ce3199ac16206c958518b968c239529900",
+    "0c02a2d596fd75d35541f37fd7805ed373ffa2780d9e94cdd47650cc711514d0",
+    "6479d7d7b19cce99a8971e85a3756ea6f1debe89b68d0a86e69cc9c6e7414d7d",
+    "64ec030272495820a69a90e76affc1d0c47377c80f267ed21fa039c40404d4c9",
+    "bc8000290bc8c8c0ebadb5c9d96dac50df8244426ef375a23cfae334e9b100c2",
+    "1b91aa4fc5e5ae49577087b2b42821ac87b863ba4de61cdccdd6b3620f587608",
+    "f26176d736b2c98959127f2c53a94ba4c3c060368eb6797ad06dd923012b53bd",
+    "539db0d62ee9e10949bac79127c082aaa0e8d001ddda9467cd8a1d05928a9b8b",
```

Comment:
Since some are 100% vulnerable and one potentially, can we please split the risk rating depending on the hash.